torch/_dynamo/bytecode_transformation.py

```diff
@@ -63,6 +63,33 @@ def create_jump_absolute(target):
     return create_instruction(inst, target=target)
 
 
+def create_dup_top():
+    if sys.version_info >= (3, 11):
+        return create_instruction("COPY", 1)
+    return create_instruction("DUP_TOP")
+
+
+def create_rot_n(n):
+    if n <= 1:
+        # don't rotate
+        return []
+
+    if sys.version_info >= (3, 11):
+        # rotate can be expressed as a sequence of swap operations
+        # e.g. rotate 3 is equivalent to swap 3, swap 2
+        return [create_instruction("SWAP", i) for i in range(n, 1, -1)]
+
+    # ensure desired rotate function exists
+    if sys.version_info < (3, 8) and n >= 4:
+        raise AttributeError(f"rotate {n} not supported for Python < 3.8")
```

Comment:
This used to work on all python versions. Why are we restricting it here?

Reply:
Ho this is caught by the caller... That feel bad, but ok.

Reply:
I'll make a comment to clarify